### PR TITLE
Fix `Start-Job` to check the existence of working directory using the PowerShell way

### LINKED ISSUE #18618: Starting background job in a UNC path with `&` returns error
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Since Powershell 7.3 i can't start a background job via the `&` shorthand with an UNC path as working directory. 
It's working fine when using `Start-Job { }`.

### Expected behavior

```console
PS Microsoft.PowerShell.Core\FileSystem::\\localhost\Share> $null &

Id     Name            PSJobTypeName   State         HasMoreData     Location             Command
--     ----            -------------   -----         -----------     --------             -------
1      Job1            BackgroundJob   Running       True            localhost            Microsoft.PowerShell.Man…
```

### Actual behavior

```console
PS Microsoft.PowerShell.Core\FileSystem::\\localhost\Share> $null &
Start-Job: Cannot find the WorkingDirectory path Microsoft.PowerShell.Core\FileSystem::\\localhost\Share.
```

### Error details

```console
Exception             :
    Type       : System.IO.DirectoryNotFoundException
    TargetSite :
        Name          : ThrowTerminatingError
        DeclaringType : System.Management.Automation.MshCommandRuntime, System.Management.Automation, Version=7.3.0.500, Culture=neutral,
PublicKeyToken=31bf3856ad364e35
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message    : Cannot find the WorkingDirectory path Microsoft.PowerShell.Core\FileSystem::\\localhost\Share.
    Source     : System.Management.Automation
    HResult    : -2147024893
    StackTrace :
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
CategoryInfo          : InvalidOperation: (:) [Start-Job], DirectoryNotFoundException
FullyQualifiedErrorId : DirectoryNotFoundException,Microsoft.PowerShell.Commands.StartJobCommand
InvocationInfo        :
    MyCommand        : Start-Job
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : -1
    Line             : $null &
    PositionMessage  : At line:1 char:1
                       + $null &
                       + ~~~~~
    InvocationName   : Start-Job
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
```

### Environment data

```powershell
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

Powershell 7 3

Powershell 7 2

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix #18618

Fix `Start-Job` to check the existence of working directory using the PowerShell way.
Use `InvokeProvider.Item.IsContainer` instead of `Directory.Exists` to verify if the specified working directory path exists.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] NoneMake sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/Po
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/commands/StartJob.cs [hunk 1]
```diff
@@ -601,7 +601,7 @@ protected override void BeginProcessing()
                 ThrowTerminatingError(errorRecord);
             }
 
-            if (WorkingDirectory != null && !Directory.Exists(WorkingDirectory))
+            if (WorkingDirectory != null && !InvokeProvider.Item.IsContainer(WorkingDirectory))
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/commands/StartJob.cs:604]: Please fix failing test for whitespace working directory.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/commands/StartJob.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/remoting/commands/StartJob.cs:604]: The test was fixed.

**Review by iSazonov — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]
LGTM

**Pull request merged**